RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate RoomItem to Hooks

## Proposed changes

Migrate RoomItem to hooks

## Issue(s)	


## How to test or reproduce

Check if the status is changing correctly.
Check the gestures, for right or left.

## Screenshots

## Types of changes




## Checklist



## Further comments
